bot/rules/mentions.py

```diff
@@ -15,10 +17,29 @@ async def apply(
 
     total_recent_mentions = sum(
         not user.bot
+        and msg.author != user
         for msg in relevant_messages
         for user in msg.mentions
     )
 
+    # we don't want to include the mentions that the author mentioned with a reply
+    for msg in relevant_messages:
+        # break the loop to save processing if total_recent_mentions is 0
+        # additionally, ensures that we don't find a mention that wasn't originally there.
+        if not total_recent_mentions:
+            break
+        if not msg.reference:
+            continue
+        resolved = msg.reference.resolved
+        if isinstance(resolved, DeletedReferencedMessage):
+            # can't figure out the author
+            continue
+        if not resolved:
+            ref = msg.reference
+            resolved = await bot.instance.get_partial_messageable(ref.channel_id).fetch_message(ref.message_id)
+        if resolved.author in msg.mentions:
+            total_recent_mentions -= 1
```

Comment:
It could be that the user is actually mentioned in the message though.
I may have set you on the wrong direction by saying "subtract", but it might be simpler to find all user mentions with regex.

Reply:
As discussed on [discord](https://discord.com/channels/267624335836053506/635950537262759947/921471985643569192), we're keeping this subtraction